src/aws-cpp-sdk-core/source/utils/crypto/CryptoBuf.cpp

```diff
@@ -289,27 +289,24 @@ namespace Aws
                         m_isFinalized = true;
                     }
 
-                    if (m_cipher)
+                    if(cryptoBuf.GetLength())
                     {
-                        if(cryptoBuf.GetLength())
+                        //allow mid block decryption. We have to decrypt it, but we don't have to write it to the stream.
+                        //the assumption here is that tellp() will always be 0 or >= 16 bytes. The block offset should only
+                        //be the offset of the first block read.
+                        size_t len = cryptoBuf.GetLength();
+                        size_t blockOffset = m_stream.tellp() > m_blockOffset ? 0 : m_blockOffset;
```

Comment:
https://en.cppreference.com/w/cpp/io/basic_ostream/tellp
> If fail()==true, returns pos_type(-1)

I'd suggest checking first that the stream is in a valid state before comparing against the m_blockOffset (although it is a `int16_t`, still I'd prefer to check the stream state)